.Rprofile

```diff
@@ -1,47 +1,45 @@
-# Set renv profile base on R version.
-.get_dependencies <- function(project_dir) {
-  admdev_loc <- find.package("admiraldev", lib.loc = .libPaths(), quiet = TRUE)
-  adm_dev_suggests <- if (length(admdev_loc) != 0) {
-    renv:::renv_dependencies_discover_description(admdev_loc, fields = c("Depends", "Imports", "LinkingTo", "Suggests"))
-  } else {
-    data.frame(Packages = character(0))
-  }
-  suggests_packages <- renv:::renv_dependencies_discover_description(project_dir, fields = "Suggests")
+if (Sys.getenv("GITHUB_ACTIONS") == "") {
+  # Set renv profile base on R version.
+  .get_dependencies <- function(project_dir) {
+    admdev_loc <- find.package("admiraldev", lib.loc = .libPaths(), quiet = TRUE)
+    adm_dev_suggests <- if (length(admdev_loc) != 0) {
+      renv:::renv_dependencies_discover_description(admdev_loc, fields = c("Depends", "Imports", "LinkingTo", "Suggests"))
+    } else {
+      data.frame(Packages = character(0))
+    }
+    suggests_packages <- renv:::renv_dependencies_discover_description(project_dir, fields = "Suggests")
 
-  packages <- names(
-    renv:::renv_package_dependencies(
-      unique(c(
-        project_dir,
-        adm_dev_suggests[["Package"]],
-        suggests_packages[["Package"]],
-        c("staged.dependencies", "renv", "styler")
-      ))
+    packages <- names(
+      renv:::renv_package_dependencies(
+        unique(c(
+          project_dir,
+          adm_dev_suggests[["Package"]],
+          suggests_packages[["Package"]],
+          c("staged.dependencies", "renv", "styler")
+        ))
+      )
     )
-  )
-  packages[!(packages %in% c("admiral", "admiraldev", "admiralci", "pharmaversesdtm", getwd()))]
-}
+    packages[!(packages %in% c("admiral", "admiraldev", "admiralci", "pharmaversesdtm", getwd()))]
+  }
 
-options(renv.snapshot.filter = .get_dependencies)
+  options(renv.snapshot.filter = .get_dependencies)
 
-.renv_profile <- paste(R.version$major, substr(R.version$minor, 1, 1), sep = ".")
-if (!file.exists("./renv/profile")) {
-  if (.renv_profile %in% c("4.1", "4.2", "4.3")) {
-    message("Set renv profile to `", .renv_profile, "`")
-    Sys.setenv("RENV_PROFILE" = .renv_profile)
+  .renv_profile <- paste(R.version$major, substr(R.version$minor, 1, 1), sep = ".")
+  if (!file.exists("./renv/profile")) {
+    if (.renv_profile %in% c("4.1", "4.2", "4.3")) {
+      message("Set renv profile to `", .renv_profile, "`")
+      Sys.setenv("RENV_PROFILE" = .renv_profile)
+    } else {
+      message("This repository do not contains the renv profile for your R version.")
+    }
   } else {
-    message("This repository do not contains the renv profile for your R version.")
+    message(
+      "Using renv profile from `renv/profile` file.\n",
+      "The `", readLines("./renv/profile"), "` profile will be used."
+    )
```

Comment:
Isn't this going to search for something that doesn't exist anymore??

Reply:
yes .. actually it's well configured for github actions but I didn't check for local config, but the question is what would you like to have exactly for local R sessions ? (I am not sure to fully understand how you want to work locally - we will have both cases people using renv and some others using devtools .. ? (I am not even sure we need to customize this file anymore to be honest) .. what do you think @cicdguy ?